Content.Server/_RMC14/Rules/CMDistressSignalRuleSystem.cs

```diff
@@ -1332,10 +1335,20 @@ public void SetFriendlyHives(EntityUid hive)
             _hive.SetHive(weeds, hive);
         }
 
-        var tunnelQuery = EntityQueryEnumerator<XenoTunnelComponent>();
-        while (tunnelQuery.MoveNext(out var uid, out _))
+    }
+    /// <summary>
+    /// Spawn tunnels from marker entities and ensure they are set to the correct hive
+    /// </summary>
+    public void SpawnTunnelsFromMarkers(EntityUid hive)
+    {
+        var tunnelSpawnerQuery = EntityQueryEnumerator<XenoTunnelSpawnerComponent>();
+        while (tunnelSpawnerQuery.MoveNext(out var uid, out var tunnelSpawnerComp))
         {
-            _hive.SetHive(uid, hive);
+            if (tunnelSpawnerComp.TunnelName is null)
+            {
+                continue;
+            }
```

Comment:
This should ideally just generate the name on its own like 13 does, which uses the area name and coordinates:
`Southwest Abandoned Mining Caves (9, 16) Alpha`